dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -117,18 +117,19 @@ internal KernelFunction(string name, string description, IReadOnlyList<KernelPar
     /// The <see cref="PromptExecutionSettings"/> to use with the function. These will apply unless they've been
     /// overridden by settings passed into the invocation of the function.
     /// </param>
-    /// <param name="properties">Properties/metadata associated with the function itself rather than its parameters and return type.</param>
-    internal KernelFunction(string name, string? pluginName, string description, IReadOnlyList<KernelParameterMetadata> parameters, KernelReturnParameterMetadata? returnParameter = null, Dictionary<string, PromptExecutionSettings>? executionSettings = null, ReadOnlyDictionary<string, object?>? properties = null)
+    /// <param name="additionalMetadata">Properties/metadata associated with the function itself rather than its parameters and return type.</param>
+    internal KernelFunction(string name, string? pluginName, string description, IReadOnlyList<KernelParameterMetadata> parameters, KernelReturnParameterMetadata? returnParameter = null, Dictionary<string, PromptExecutionSettings>? executionSettings = null, ReadOnlyDictionary<string, object?>? additionalMetadata = null)
     {
         Verify.NotNull(name);
         Verify.ParametersUniqueness(parameters);
 
-        this.Metadata = new KernelFunctionMetadata(name, properties)
+        this.Metadata = new KernelFunctionMetadata(name)
         {
             PluginName = pluginName,
             Description = description,
             Parameters = parameters,
             ReturnParameter = returnParameter ?? KernelReturnParameterMetadata.Empty,
+            AdditionalProperties = additionalMetadata ?? KernelFunctionMetadata.s_emptyDictionary,
```

Comment:
Should this be named `AdditionalMetadata`?

Reply:
I think outside of the metadata class AdditionalMetadata makes sense, since it's clarifying where the data will end up.  However inside the metadata class the word metadata is unnecessary and properties make more sense.